### PR TITLE
fix(types): four-oh-four props, global i18next definition for returnNull 

### PR DESCRIPTION
**Pull request by mateoguzmana**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Removing non-needed prop types for the `FourOhFour` component to fix a ts-ignore.
Fixing issue with `string | undefined` by adding a proper global i18next definition.

### CODE REVIEW

**Review by huyenltnguyen — DISMISSED** [review]

File: client/src/pages/404.tsx [hunk 1]
```diff
@@ -14,8 +14,7 @@ function FourOhFourPage(): JSX.Element {
       {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
       {/* @ts-ignore */}
       <ShowProfileOrFourOhFour path={withPrefix('/:maybeUser')} />
-      {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
-      {/* @ts-ignore */}
+
       <FourOhFour default={true} />
```
[the remark this answers is not in the document]
Comment (mateoguzmana) [reply client/src/pages/404.tsx]: That makes sense!

File: client/src/components/FourOhFour/index.tsx [hunk 2]
```diff
@@ -9,13 +9,18 @@ import { Spacer } from '../helpers';
...
+
+const FourOhFour = (_props: FourOhFourProps): JSX.Element => {
   const { t } = useTranslation();
   const quote = randomQuote();
+
   return (
     <div className='notfound-page-wrapper'>
       <Helmet title={t('404.page-not-found') + '| freeCodeCamp'} />
-      <img alt={t('404.not-found')} src={notFoundLogo} />
+      <img alt={t('404.not-found') || ''} src={notFoundLogo} />
```
[the remark this answers is not in the document]
Comment (mateoguzmana) [reply client/src/components/FourOhFour/index.tsx]: Good one, thanks! I've fixed it properly with the global.

File: client/src/pages/404.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/src/pages/404.tsx]: While the `FourOhFour` component doesn't use the prop, reach-router does. Without it the component will not be rendered.

Instead, what we can do here is make `FourOhFour` take `RouteComponentProps` (from `@reach/router`) as its argument. i.e. something like this:

```ts
const FourOhFour = (_props: RouteComponentProps): JSX.Element => {
```

[the remark this answers is not in the document]
Comment (mateoguzmana) [reply client/src/pages/404.tsx]: Thank you both for the clarification, I've put back the `default` and added the props declaration as suggested.

**Review by ojeytonwilliams — APPROVED** [review]
Looks good @mateoguzmana, :+1: 

Thanks for helping with the i18next types, @huyenltnguyen that should fix a lot of issues!

**Review by Sembauke — APPROVED** [review]

**Pull request merged**